RealmSwift/MongoClient.swift

```diff
@@ -0,0 +1,506 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2020 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import Foundation
+import Realm
+import Realm.Private
+
+/**
+ * The `MongoClient` enables reading and writing on a MongoDB database via the Realm Cloud service.
+ *
+ * It provides access to instances of `MongoDatabase`, which in turn provide access to specific
+ * `MongoCollection`s that hold your data.
+ *
+ * - Note:
+ * Before you can read or write data, a user must log in.
+ *
+ * - SeeAlso:
+ * `RealmApp`, `MongoDatabase`, `MongoCollection`
+ */
+public typealias MongoClient = RLMMongoClient
+
+/**
+ * The `MongoDatabase` represents a MongoDB database, which holds a group
+ * of collections that contain your data.
+ *
+ * It can be retrieved from the `MongoClient`.
+ *
+ * Use it to get `MongoCollection`s for reading and writing data.
+ *
+ * - Note:
+ * Before you can read or write data, a user must log in`.
+ *
+ * - SeeAlso:
+ * `MongoClient`, `MongoCollection`
+ */
+public typealias MongoDatabase = RLMMongoDatabase
+
+/// Options to use when executing a `find` command on a `MongoCollection`.
+public typealias FindOptions = RLMFindOptions
+
+extension FindOptions {
+
+    /// Limits the fields to return for all matching documents.
+    public var projection: Document? {
+        get {
+            guard let value = ObjectiveCSupport.convert(object: __projection) else {
+                return nil
+            }
+            return value.documentValue
+        }
+        set {
+            if let value = newValue {
```

Comment:
This doesn't let you change the value from non-nil to nil. Something like `__projection = newValue.map(AnyBSON.init).map(ObjectiveCSupport.convert)` should work to handle that correctly as a one-liner.